packages/rest/src/__tests__/unit/rest.server/rest.server.open-api-spec.unit.ts

```diff
@@ -393,6 +408,137 @@ describe('RestServer.getApiSpec()', () => {
     expect(spec.info).to.eql(EXPECTED_SPEC_INFO);
   });
 
+  context('options', () => {
+    it('disables consolidator if consolidate is set to false', async () => {
+      const options: {rest: RestServerConfig} = {
+        rest: {openApiSpec: {consolidate: false}},
+      };
+      options.rest = givenHttpServerConfig(options.rest);
+      app = new Application(options);
+      app.component(RestComponent);
+      server = await app.getServer(RestServer);
+      await server.start();
+
+      server.api(
+        anOpenApiSpec()
+          .withOperation(
+            'get',
+            '/',
+            anOperationSpec().withResponse(200, {
+              description: 'Example',
+              content: {
+                'application/json': {
+                  schema: {
+                    title: 'loopback.example',
+                    properties: {
+                      test: {
+                        type: 'string',
+                      },
+                    },
+                  },
+                },
+              },
+            }),
+          )
+          .build(),
+      );
+
+      const EXPECTED_SPEC = anOpenApiSpec()
+        .withOperation(
+          'get',
+          '/',
+          anOperationSpec().withResponse(200, {
+            description: 'Example',
+            content: {
+              'application/json': {
+                schema: {
+                  title: 'loopback.example',
+                  properties: {
+                    test: {
+                      type: 'string',
+                    },
+                  },
+                },
+              },
+            },
+          }),
+        )
+        .build();
+
+      const spec = await server.getApiSpec();
```

Comment:
@dougal83 This will remove `spec.paths` as we try to build openapi spec from the handler. See https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/rest.server.ts#L800. 

You probably have to add a route with the expected api spec.

Reply:
@dougal83 The following code works for me.

1. We need to call server.route() to set up a real route
2. server.route() needs to be called before `start`.

```ts
  context('options', () => {
    it('disables consolidator if consolidate is set to false', async () => {
      const options: {rest: RestServerConfig} = {
        rest: {openApiSpec: {consolidate: false}},
      };
      options.rest = givenHttpServerConfig(options.rest);
      app = new Application(options);
      app.component(RestComponent);
      server = await app.getServer(RestServer);

      const EXPECTED_SPEC = anOpenApiSpec()
        .withOperation(
          'get',
          '/',
          anOperationSpec().withResponse(200, {
            description: 'Example',
            content: {
              'application/json': {
                schema: {
                  title: 'loopback.example',
                  properties: {
                    test: {
                      type: 'string',
                    },
                  },
                },
              },
            },
          }),
        )
        .build();

      server.route('get', '/', EXPECTED_SPEC.paths['/'].get, () => {});

      await server.start();
      const spec = await server.getApiSpec();
      expect(spec).to.eql(EXPECTED_SPEC);

      await server.stop();
    });

    it('runs consolidator if consolidate is set to true', async () => {
      const options: {rest: RestServerConfig} = {
        rest: {openApiSpec: {consolidate: true}},
      };
      options.rest = givenHttpServerConfig(options.rest);
      app = new Application(options);
      app.component(RestComponent);
      server = await app.getServer(RestServer);

      const EXPECTED_SPEC = anOpenApiSpec()
        .withOperation(
          'get',
          '/',
          anOperationSpec().withResponse(200, {
            description: 'Example',
            content: {
              'application/json': {
                schema: {
                  $ref: '#/components/schemas/loopback.example',
                },
              },
            },
          }),
        )
        .withComponents(
          aComponentsSpec().withSchema('loopback.example', {
            title: 'loopback.example',
            properties: {
              test: {
                type: 'string',
              },
            },
          }),
        )
        .build();

      server.route(
        'get',
        '/',
        anOperationSpec()
          .withResponse(200, {
            description: 'Example',
            content: {
              'application/json': {
                schema: {
                  title: 'loopback.example',
                  properties: {
                    test: {
                      type: 'string',
                    },
                  },
                },
              },
            },
          })
          .build(),
        () => {},
      );

      await app.start();
      const spec = await server.getApiSpec();
      expect(spec).to.eql(EXPECTED_SPEC);

      await app.stop();
    });
  });
```

Reply:
BTW, this is not really a unit test anymore. We probably should move it to `integration`.

Reply:
thank you @raymondfeng . In `getApiSpec`, [`spec.paths`](https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/rest.server.ts#L800) and [`spec.components.schemas`](https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/rest.server.ts#L803) are not preserved, is it intended? 
I remember we discussed this before, if the paths and schemas should be kept I can submit a PR to fix it. 
(the discussion here is not related to PR I can open a new issue if we need more opinions)

Reply:
For the rest configurations:
the unit tests are added in [__tests__/unit/rest.server/rest.server.unit.ts](https://github.com/strongloop/loopback-next/blob/a81ce7e1193f7408d30d984d0c3ddcec74f7c316/packages/rest/src/__tests__/unit/rest.server/rest.server.unit.ts#L78-L94)
and integration tests are in [__tests__/integration/rest.server.integration.ts](https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/__tests__/integration/rest.server.integration.ts#L851)
links point to the existing options tests.

Reply:
@raymondfeng Ah OK, I didn't understand the getApiSpec() implementation.  Could you please take a look at the code below? I want to test the rest config is being applied.

```ts
     server.api(INPUT_SPEC);

      server.OASEnhancer.spec = await server.get<OpenApiSpec>(
        RestBindings.API_SPEC,
      );
      await server.OASEnhancer.applyEnhancerByName('consolidate');
      const spec = await server.get<OpenApiSpec>(RestBindings.API_SPEC);
      expect(spec).to.eql(INPUT_SPEC);
```

Reply:
@raymondfeng Thank you very much for your guidance! 👍 

@jannyHou Thank you for your help!  Could you please check that the tests are located in the correct place now?  I've no formal training on tests so I'm really just hacking.